src/lib/logical_query_plan/lqp_utils.cpp

```diff
@@ -84,10 +82,50 @@ void lqp_find_subplan_roots_impl(std::vector<std::shared_ptr<AbstractLQPNode>>&
   });
 }
 
+void recursively_collect_subquery_expressions_by_lqp(
+    SubqueryExpressionsByLQP& subquery_expressions_by_lqp, const std::shared_ptr<AbstractLQPNode>& node,
+    std::unordered_set<std::shared_ptr<AbstractLQPNode>>& visited_nodes) {
+  if (!node) return;
+  if (!visited_nodes.emplace(node).second) return;
+
+  for (const auto& expression : node->node_expressions) {
+    visit_expression(expression, [&](const auto& sub_expression) {
+      const auto subquery_expression = std::dynamic_pointer_cast<LQPSubqueryExpression>(sub_expression);
+      if (!subquery_expression) return ExpressionVisitation::VisitArguments;
+
+      for (auto& [lqp, subquery_expressions] : subquery_expressions_by_lqp) {
```

Comment:
const?

Reply:
Not possible because we modify `subquery_expressions`